### PR TITLE
implemented real estate scenario

### PR DESCRIPTION
**Pull request by puahisrael**
I (attempted to) implement the real estate biz scenario. Please review.
Thanks!

### CODE REVIEW

File: 1 table.sql [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    SqrFootOfHouse int not null constraint c_House_sqr_foot_of_house_must_be_greater_than_0 check(SqrFootOfHouse > 0),
+    SqrFootOfLot int not null constraint c_House_sqr_foot_of_lot_must_be_greater_than_0 check(SqrFootOfLot > 0),
+    NumOfBath decimal(3,1) not null constraint c_House_num_of_bath_cannot_be_negative check(NumOfBath >= 0),
+    NumOfBed decimal(3,1) not null constraint c_House_num_of_bed_cannot_be_negative check(NumOfBed >= 0),
+    HouseOwner varchar(25) not null constraint c_House_house_owner_cannot_be_blank check(HouseOwner <> ''),
+    Realtor varchar(25) not null constraint c_House_realtor_cannot_be_blank check(Realtor <> ''),
+    Client varchar(25) not null constraint c_House_client_cannot_be_blank check(Client <> ''),
+    DateOnMarket date not null,
+    Buyer varchar(50) null constraint c_House_buyer_cannot_be_blank check(Buyer <> ''),
+    DateSold date null constraint c_House_date_sold_must_be_before_current_date check(DateSold < getdate()),
```
Comment (AvigailGellis) [inline 1 table.sql:20]: Allow a house to be sold on the current date.

File: 1 table.sql [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    AskingPrice decimal(9,2) null,
```
Comment (AvigailGellis) [inline 1 table.sql:21]: The constraint that price has to be between 100 k and 9.9 million should be on this column as well.

File: 1 table.sql [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    SellingPrice decimal(9,2) null constraint c_House_selling_price_must_be_between_100000_and_9999999 check(SellingPrice between 100000 and 9999999),
```
Comment (AvigailGellis) [inline 1 table.sql:22]: 9.9 million would be 9,900,000

File: 1 table.sql [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    InContract bit null, 
+
+    constraint c_House_date_sold_cannot_be_before_date_on_market check(DateSold >= DateOnMarket),
+    constraint c_House_selling_price_cannot_be_less_than_asking_price check(SellingPrice >= AskingPrice)
+)
+
```
Comment (AvigailGellis) [inline 1 table.sql:28]: Add in column for dateinserted, as explained in the spec.

File: 1 table.sql [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    NumOfBed decimal(3,1) not null constraint c_House_num_of_bed_cannot_be_negative check(NumOfBed >= 0),
+    HouseOwner varchar(25) not null constraint c_House_house_owner_cannot_be_blank check(HouseOwner <> ''),
+    Realtor varchar(25) not null constraint c_House_realtor_cannot_be_blank check(Realtor <> ''),
+    Client varchar(25) not null constraint c_House_client_cannot_be_blank check(Client <> ''),
+    DateOnMarket date not null,
+    Buyer varchar(50) null constraint c_House_buyer_cannot_be_blank check(Buyer <> ''),
+    DateSold date null constraint c_House_date_sold_must_be_before_current_date check(DateSold < getdate()),
+    AskingPrice decimal(9,2) null,
+    SellingPrice decimal(9,2) null constraint c_House_selling_price_must_be_between_100000_and_9999999 check(SellingPrice between 100000 and 9999999),
+    InContract bit null,
```
Comment (AvigailGellis) [inline 1 table.sql:23]: Nice! This column can be done as a computed column.

File: 1 table.sql (hunk 2 above)
Comment (AvigailGellis) [reply 1 table.sql:21]: AskingPrice should not allow nulls, there needs to be an asking price.

File: 1 table.sql [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    HouseType varchar(20) not null constraint c_House_house_type_must_be_bi_level_colonial_ranch_split_level_duplex_townhouse_apartment_or_vacant_land check(HouseType in('bi-level','colonial','ranch','split-level','duplex','townhouse','apartment','vacant land' )),
+    SqrFootOfHouse int not null constraint c_House_sqr_foot_of_house_must_be_greater_than_0 check(SqrFootOfHouse > 0),
+    SqrFootOfLot int not null constraint c_House_sqr_foot_of_lot_must_be_greater_than_0 check(SqrFootOfLot > 0),
+    NumOfBath decimal(3,1) not null constraint c_House_num_of_bath_cannot_be_negative check(NumOfBath >= 0),
+    NumOfBed decimal(3,1) not null constraint c_House_num_of_bed_cannot_be_negative check(NumOfBed >= 0),
+    HouseOwner varchar(25) not null constraint c_House_house_owner_cannot_be_blank check(HouseOwner <> ''),
+    Realtor varchar(25) not null constraint c_House_realtor_cannot_be_blank check(Realtor <> ''),
+    Client varchar(25) not null constraint c_House_client_cannot_be_blank check(Client <> ''),
+    DateOnMarket date not null,
+    Buyer varchar(50) null constraint c_House_buyer_cannot_be_blank check(Buyer <> ''),
```
Comment (AvigailGellis) [inline 1 table.sql:19]: Buyer should not allow nulls, rather insert a blank when necessary.

File: 3 reports.sql [hunk 7]
```diff
@@ -0,0 +1,19 @@
+use RealEstateDb
+
+--1) I want a report of all the houses sorted by the block and then by the town/city
```
Comment (AvigailGellis) [inline 3 reports.sql:3]: It seems like from this question that it is important to split up the house address into 2 columns: House number and Street

File: 3 reports.sql [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+select * from House h
```
Comment (AvigailGellis) [inline 3 reports.sql:4]: Formatting tip: Questions 1 and 2 the from clause belong on the next line.

**Review by AvigailGellis — COMMENTED** [review]
Great job! Please see comments and resubmit.

File: 1 table.sql (hunk 1 above)
Comment (puahisrael) [reply 1 table.sql]: I modified this constraint.

File: 1 table.sql (hunk 2 above)
Comment (puahisrael) [reply 1 table.sql]: I fixed these mistakes.

File: 1 table.sql (hunk 3 above)
Comment (puahisrael) [reply 1 table.sql]: Yeah that's true... did it in previous commit

File: 1 table.sql (hunk 4 above)
Comment (puahisrael) [reply 1 table.sql:29]: K did it...

File: 1 table.sql (hunk 5 above)
Comment (puahisrael) [reply 1 table.sql]: I don't know if this is what you meant but I tried...

File: 1 table.sql (hunk 6 above)
Comment (puahisrael) [reply 1 table.sql]: Done...

File: 3 reports.sql (hunk 7 above)
Comment (puahisrael) [reply 3 reports.sql:3]: Okay, great

File: 3 reports.sql (hunk 8 above)
Comment (puahisrael) [reply 3 reports.sql]: Yeah, sorry about that.
I hope everything is better now...

**Pull request closed without merging**